### DISCUSSION TITLE
Audio Blob file not uploading

### DISCUSSION THREAD

**Discussion by thisishasan** [livewire/livewire#3922] [7. Bugs (archived)] [answered]
The blob generating from Media Stream Recording API is unable to upload on the server via component. Please help!

https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API

**Comment by jonneroelofs** [comment]
Kind of hard to tell what the problem is without seeing the code and/or errors you are getting in the console/networks tab. 

What I would recommend is to make sure you have the WithFileUploads trait on your component and that you checkout this part of the docs: 
https://laravel-livewire.com/docs/2.x/file-uploads#js-api

**Reply by thisishasan** [reply]
[https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API](url)

I am using JS Media Recorder library.

**Reply by thisishasan** [reply]
in the frontend I am using javascript to assign the blob like this:
`@this.blob = blob`

in the component backend code when i debug it like this; it prints null
`dd($this->blob)`

**Reply by thisishasan** [reply]
However I am using WithFileUploads trait.. this works fine for all other uploads.. just for blob its not working

**Comment by jonneroelofs** [comment, marked as answer]
In the docs it states you have to do @this.upload('blob', blob)

**Reply by thisishasan** [reply]
thanks its working now
